loadtest/control/simulcontroller/actions.go

```diff
@@ -1410,6 +1423,15 @@ func (c *SimulController) initialJoinTeam(u user.User) control.UserActionRespons
 		return c.joinTeam(c.user)
 	}
 
+	if c.user.Store().FeatureFlags()["WebSocketEventScope"] {
+		// Setting the active thread to empty to allow the optimization to kick in early.
+		// We don't do this in the webapp to keep the code simple, but it's okay to do this in load-test
+		// to magnify the boost.
```

Comment:
When in doubt I'd probably want the load-test simulation to perform worse than webapp to be on the safe side. I'd be okay with keeping this to prove the ideal improvements though.

Reply:
The problem is our simulation is not close enough to the webapp. We don't differentiate between loading a thread from RHS or from global thread view. It's all just a loadThread action. Until we differentiate that, we'll have to approximate somehow.

Reply:
But are we making things more performant here with this? Or am I misunderstanding the goal?

Reply:
Correct, we are making it more performant here.

Reply:
That's what worries me (and what I think concerned Claudio as well). Is the performance improvement in here really needed?

Reply:
I talked about this earlier in this comment: https://github.com/mattermost/mattermost-load-test-ng/pull/678#discussion_r1459290297

Until we do that, we have to approximate somehow. And if we choose not to set the active thread at all, then there will be no improvements at all, so it will be impossible to measure the impact of the change.

Reply:
Ok, let's go with it then! Thank you for the patience :)